File: Artifact 3/Enhanced Code - Artifact 3/Grocery-tracker-backend/src/main/java/com/carmen/GroceryTracker/Model/Category.java
/**
 * Grocery tracker backend - category model class
 * Version: 1.0
 * Developer: Carmen Mosquera
 * Description - App: This application analyzes and tracks the frequency a product is purchased in a day.
 * This application uses mySQL database to store and retrieve product information.
 * Description - Class: This class serves as model for the category entity.
 */

package com.carmen.GroceryTracker.Model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import java.util.Set;

@Entity
@Table(name = "categories")
public class Category {

    //ID
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    //NAME
    @Column(nullable = false)
    private String name;

    // PRODUCTS (One-to-Many relationship)
    @OneToMany(mappedBy = "productCategory", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonManagedReference(value = "product-category")  // To prevent circular reference in serialization
    private Set<Product> products;



    ////////////  Constructors
    public Category() {}

    public Category(String name) {
        this.name = name;
    }

    //////////////// GETTERS AND SETTERS
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Set<Product> getProducts() {
        return this.products;
    }

    public void setProducts(Set<Product> products) {
        this.products = products;
    }
}

File: Grocery Tracker Application/Enhanced Full Stack version/Grocery-Tracker-Frontend/src/app/model/category.ts
/**
 * Grocery tracker frontend - category model component
 * Version: 1.0
 * Developer: Carmen Mosquera
 * Description - App: This application analyzes and tracks the frequency a product is purchased in a day.
 * This application uses mySQL database to store and retrieve product information.
 * Description - Class: This component serves as model for the category entity.
 */
import {ProductDTO} from './product';


export class CategoryDTO {
  id: number;
  name!: string;
  products!: ProductDTO[];


  constructor(name: string) {
    this.id = 0;
    this.name = name;
    this.products = [] ;
  }
}
